cmd/platform/jobserver.go

```diff
@@ -46,7 +46,7 @@ func jobserverCmdF(cmd *cobra.Command, args []string) {
 		a.Jobs.StartSchedulers()
 	}
 
-	var signalChan chan os.Signal = make(chan os.Signal)
```

Comment:
In addition to being unnecessarily long, this line causes a subtle race condition. The signal package will never block when notifying a channel. So if a signal was received before we parked ourselves on line 51, we'd never know. It's pretty inconsequential in our case, but it's a good practice and gives us the guarantee that once `Notify` returns, we'll receive any future signals.

> Package signal will not block sending to c: the caller must ensure that c has sufficient buffer space to keep up with the expected signal rate. For a channel used for notification of just one signal value, a buffer of size 1 is sufficient.

– https://golang.org/pkg/os/signal/#Notify

Reply:
Nice catch